pennylane/tape/qnode.py

```diff
@@ -402,8 +402,13 @@ def construct(self, args, kwargs):
             # controlled rotations aren't supported by the parameter-shift rule
             stop_at = set(self.device.operations) - {"CRX", "CRZ", "CRY", "CRot"}
 
-        # expand out the tape, if any operations are not supported on the device
-        if not {op.name for op in self.qtape.operations}.issubset(stop_at):
+        # pylint: disable=protected-access
+        obs_on_same_wire = len(self.qtape._repeated_observables) > 0
+        ops_not_supported = not {op.name for op in self.qtape.operations}.issubset(stop_at)

```

Comment:
Right, `stop_at` is the operations available on the device*, and we know we can stop there since they are supported.
We then loop through all the requested operations (in `self.qtape.operations`) and check if they are within the supported operations, if not, we'll need to expand.

\* we also change `stop_at` to remove the controlled rotations, due to the bug in parameter shift.